src/provider.rs

```diff
@@ -15,14 +15,14 @@ pub trait Provider {
     /// See [RFC 6749, section 3.1](http://tools.ietf.org/html/rfc6749#section-3.1).
     ///
     /// Note: likely to become an associated constant.
-    fn auth_uri() -> &'static str;
+    fn auth_uri(&self) -> &str;
```

Comment:
At least in my implementation the Provider impl now stores auth_uri and token_uri internally and just shares those strings as_ref. My clients are just making Hyper::Uri's out of these strings anyway - no need to clone them.

If we are talking about the signature though, it would be really nice if these functions could return actual hyper::Uri's or url::Url's. The problem is there's no way to const either type since they are drop and all their constructors give back Results from parsing. It would require upstream support in hyper / url / maybe the new http crate.